### PR TITLE
Resolves #268 #281 & Update french stringtable

### LINKED ISSUE #268: Player restrained can open vehicle inventory
### Expected behaviour
When a player is restrained and in a vehicle, he can not open the virtual invetory or gear inventory, if the vehicle is open

### Actual behaviour
When a player is restrained and in a vehicle, he can open the virtual invetory or gear inventory, if the vehicle is open

### Steps to reproduce the behaviour 
1. Connect on COP slot
2. Restrain player
3. Put player in the vehicle and dont lock the vehicle
4. The player restrained, open de vertual inventory with "T" Key and gear inventory with keyobard key.

### My Solution

Add:
```
if (player getVariable "restrained") exitWith {hint localize "STR_NOTF_isrestrained";}; 
if (player getVariable "playerSurrender") exitWith {hint localize "STR_NOTF_surrender";}; // For player surrender and near vehicle.
```
in
```
Framework/Altis_Life.Altis/core/vehicle/fn_vehInventory.sqf
```
Like this:

> http://pastebin.com/JxNg1aPw

---------------------------------------------------
Mission version: 5.0.0

Launcher version: latest

Game version: ~latest~ 1.66

Branch: ~master~ main -- Edit: @Jawshy

### PR DESCRIPTION
**Pull request by Dim2ke**
Resolves #268 #281
Update french stringTable

### Changes proposed in this pull request: 
- #### Resolves #268 
- #### File: fn_keyHandler.sqf
- #### Line: 137
     - Add check if player can open inventory :
          - Player restrained
          - Player surrender
          - life_isknocked
          - life_istazed

- #### Update french stringtable
- #### File: stringtable.xml
- #### Line:
     - 563: Add coma
     - 589: no diminutive for better understanding
     - 712: in this context no "quantité" but "montant" (quantité is for item and other, montant is for money)
     - 785: in this context we dont activate God but activate God mod
     - 796 in this context we dont disable God but disable God mod
     - 999: Translation that seems more coherent
     - 1065: Forget the "r" in "argent"
     - 1153: Forget the "r" "argent"
     - 1289: Add French translation
     - 1300: Add French translation
     - 1311: Add French translation
     - 1322: Add French translation
     - 1344: Add French translation
     - 1379: Best translation
     - 1845: Add French translation
     - 1922: error in spelling
     - 2062: Add French translation
     - 2875: Translation that seems more coherent
     - 9896: Rename "Pavot" in "Marijuana" because it's Marijuana
     - 9907: Rename "Pavot" in "Marijuana" because it's Marijuana
     - 12038:  Translation that seems more coherent
     - 12049: Translation that seems more coherent

- #### Resolves part of #281 
- #### File: fn_buyHouseGarage.sqf
- #### Line: 21 & 22
     - Format numbertext for price
- #### File: fn_sellHouseGarage.sqf
- #### Line: 20 & 21
     - replace variable "_price" with "_sellPrice"
     - Format numbertext for sellPrice
     - Change "STR_Garage_HouseSellMSG" with "STR_House_SellGarageMSG"

### CODE REVIEW

File: Altis_Life.Altis/core/functions/fn_keyHandler.sqf [hunk 1]
```diff
@@ -143,7 +143,7 @@ switch (_code) do {
 
     //T Key (Trunk)
     case 20: {
-        if (!_alt && !_ctrlKey && !dialog && {!life_action_inUse}) then {
+        if (!_alt && !_ctrlKey && !dialog && (!(player getVariable ["restrained",false])) && (!(player getVariable ["playerSurrender",false])) && !life_isknocked && !life_istazed && {!life_action_inUse}) then {
```
Comment (Jason2605) [inline Altis_Life.Altis/core/functions/fn_keyHandler.sqf]: Lazy Eval

**Comment by BoGuu** [comment]
@Dim2ke 

Have you seen Jason's comment?

**Comment by Dim2ke** [comment]
@BoGuu @Jason2605 
Sorry for my late return, I'll do it this weekend

**Comment by Dim2ke** [comment]
@Jason2605 @BoGuu
Hello,
I made the changes, is that good ?

File: Altis_Life.Altis/stringtable.xml [hunk 2]
```diff
@@ -560,7 +560,7 @@
         <Key ID="STR_Admin_CannotTpHere">
             <Original>You cannot teleport the player here because they are inside of a vehicle.</Original>
             <Czech>Nelze teleportovat přehrávač zde proto, že jsou uvnitř vozidla.</Czech>
-            <French>Vous ne pouvez pas TP le joueur ici car il est actuellement dans un vehicule.</French>
+            <French>Vous ne pouvez pas TP le joueur ici, car il est actuellement dans un vehicule.</French>
```
Comment (Minemega19) [inline Altis_Life.Altis/stringtable.xml]: No "," before "car".

Comment (Dim2ke) [inline Altis_Life.Altis/stringtable.xml]: I'm sure there's a comma before "car", but I hesitate between:
- The comma before and after "car"
  "Vous ne pouvez pas TP le joueur ici, car, il est actuellement dans un vehicule."
- Or the comma just before "car"
  "Vous ne pouvez pas TP le joueur ici, car il est actuellement dans un vehicule."

- Try in http://www.cordial-enligne.fr/index.php
  - Without comma
   Result: False, add comma before "car"
  - With comma after "car"
   Result: False, add comma before "car" and keep comma after "car"
  - With comma before and after "car"
   Result: True, do not add comma after "car"

Comment (Minemega19) [inline Altis_Life.Altis/stringtable.xml]: You are right.

File: Altis_Life.Altis/core/functions/fn_keyHandler.sqf [hunk 3]
```diff
@@ -143,8 +143,8 @@ switch (_code) do {
 
     //T Key (Trunk)
     case 20: {
-        if (!_alt && !_ctrlKey && !dialog && {!life_action_inUse}) then {
-            if (!(isNull objectParent player) && alive vehicle player) then {
+        if (!_alt && !_ctrlKey && !dialog &&{!life_action_inUse}) then {
+            if (!(isNull objectParent player) && alive vehicle player && (!(player getVariable ["restrained",false])) && (!(player getVariable ["playerSurrender",false])) && !life_isknocked && !life_istazed) then {
```
Comment (BoGuu) [inline Altis_Life.Altis/core/functions/fn_keyHandler.sqf]: Could you wrap this all in lazy eval?

Comment (Dim2ke) [inline Altis_Life.Altis/core/functions/fn_keyHandler.sqf]: Hello @BoGuu
I tried but I did not find how to do it, can you help me and explain to me for the next time

Comment (Jawshy) [inline Altis_Life.Altis/core/functions/fn_keyHandler.sqf]: @Dim2ke, see [Code Optimisation](https://community.bistudio.com/wiki/Code_Optimisation) [CTRL +F "lazy evaluation"].

**Comment by Dim2ke** [comment]
@BoGuu @Jawshy 
I tried, can you tell me if it's good, thanx 😄 
Also, I want to add a check for when the cop restrains a player, the players must be tazed or knocked out, i can ?

**Comment by BoGuu** [comment]
LGTM - sorry for the delay

@Jawshy

File: Altis_Life.Altis/stringtable.xml [hunk 4]
```diff
@@ -560,7 +560,7 @@
         <Key ID="STR_Admin_CannotTpHere">
             <Original>You cannot teleport the player here because they are inside of a vehicle.</Original>
             <Czech>Nelze teleportovat přehrávač zde proto, že jsou uvnitř vozidla.</Czech>
-            <French>Vous ne pouvez pas TP le joueur ici car il est actuellement dans un véhicule.</French>
+            <French>Vous ne pouvez pas TP le joueur ici, car il est actuellement dans un vehicule.</French>
```
Comment (Metalhearf) [inline Altis_Life.Altis/stringtable.xml]: "vehicule" needs an accent -> "véhicule"

File: Altis_Life.Altis/stringtable.xml [hunk 5]
```diff
@@ -1307,7 +1307,7 @@
             <Spanish>Your message cannot exceed 400 characters!</Spanish>
             <Russian>Your message cannot exceed 400 characters!</Russian>
             <German>Deine Nachricht darf maximal 400 Zeichen enthalten!!</German>
-            <French>Votre message ne peut pas dépasser 400 caractères !</French>
+            <French>Votre message ne peut pas excéder 400 caractères!</French>
```
Comment (Metalhearf) [inline Altis_Life.Altis/stringtable.xml]: We need a space before the "!" :) 
[Source](https://github.com/AsYetUntitled/Framework/pull/179#issuecomment-262789117)

File: Altis_Life.Altis/stringtable.xml (hunk 4 above)
Comment (Dim2ke) [inline Altis_Life.Altis/stringtable.xml]: Thanx 👍

File: Altis_Life.Altis/stringtable.xml (hunk 5 above)
Comment (Dim2ke) [inline Altis_Life.Altis/stringtable.xml]: You are right.
http://www.la-ponctuation.com/point-exclamation.html

**Comment by Jason2605** [comment]
Thanks.

**Pull request merged**